### ISSUE TITLE
Allow setting a network timeout value

### ISSUE THREAD

**Issue by Jenrykster** [sumup/sumup-android-sdk#234]
I've noticed that sometimes my app gets stuck on the first "Processing" screen if the network is unstable. Is there any way I can set a fixed timeout so that the user can retry the payment process? Currently, it randomly takes from 20 seconds to 3 minutes for the SDK to return the user to the previous activity.

I could also implement the timeout inside my app but I couldn't find a method that allows me to forcefully cancel the checkout.

**Comment by zaid-abdullah** [comment]
Hello @Jenrykster 
It is a bit difficult to assess the phase of the transaction process as processing "under the hood" could be doing different things depending on where we are in the process.
Would be helpful if you could provide an example video 🙏 
Also, it could be that if you can cancel the transaction from the reader at this point, that might already solve your problem as readers are connected via BT.

**Comment by Jenrykster** [comment]
It gets stuck on the first screen right after the transaction starts. I was debugging it by turning off the Android device's wifi right before the sumup activity started.
It happens before the reader is connected so the user can't cancel the transaction and return to the previous screen.

I'll try to provide an example video later.

**Comment by Jenrykster** [comment]
Sorry for the delay on providing the video. Just to be clear I turned off my internet connection right before clicking on "Cartão" (card payment) to reproduce this issue.

https://github.com/user-attachments/assets/fdf512e4-6c02-4ff4-b94f-a46b608a33d9

This is the current behavior. But the time spent on the "processing" screen varies randomly, and some of my users have been reporting very long times where they can't interact with the app.

**Comment by zaid-abdullah** [comment]
@Jenrykster thanks for the video 🙏 
Can you also please specify from the SDK API which method you are calling behind "Cartão"? (just to be explicit and not to make any assumptions

**Comment by Jenrykster** [comment]
Yeah of course, the app from the video is made using react native so the code below is part of a native module.

```java
 SumUpPayment payment = SumUpPayment.builder().total(new BigDecimal(value))
                            .currency(SumUpPayment.Currency.BRL)
                            .title(title).skipSuccessScreen().build();

SumUpAPI.checkout(activity, payment, REQUEST_CODE_PAYMENT);
```

I guess that the "processing screen" appears right after the call to `checkout`

**Comment by zaid-abdullah** [comment]
Thanks for the detailed bug report. We will look into it and return to you if any other input is needed. I appreciate 🙏

**Comment by Jenrykster** [comment]
Hi, any news on this?
